src/app/workflow/view/view.service.ts

```diff
@@ -53,10 +53,10 @@ export class ViewService {
     const snapshot: WorkflowVersion = { ...version, frozen: true };
     this.workflowsService.updateWorkflowVersion(workflow.id, [snapshot]).subscribe(
       workflowVersions => {
+        const selectedWorkflow = { ...this.workflowQuery.getActive() };
```

Comment:
1. Since line 54 is asynchronous, is there the risk that `this.workflowQuery.getActive()` returns a different workflow than the one represented by `workflow.id` on line 54?
2. Why can't you just use the `workflow` parameter from line 52?

Just asking; not sure what the answers are....

Also, could you explicitly specify the type of `selectedWorkflow`?

Reply:
1. I think you're describing a situation where someone navigates away from this workflow while the snapshot is in progress. In that case the workflowVersions might get set from a different version and display some mangled data. 
2. If we use workflow from 52 then if you snapshot, browse away, it will cause the old workflow to reappear.

I think its a good catch, though rare since snapshotting should be fast. I think the expectation would be to silently succeed when someone browses away. This could be done by guaranteeing the returned workflow id matches the one gotten via querying the store, if I understand correctly.

Reply:
>     1. I think you're describing a situation where someone navigates away from this workflow while the snapshot is in progress. In that case the workflowVersions might get set from a different version and display some mangled data.

 Yes, I was guessing at something like that, even though in practice it's a stretch to happen.

>     2. If we use workflow from 52 then if you snapshot, browse away, it will cause the old workflow to reappear.

Even if you clone it, set its workflowVersions, and call this.workflowService.setWorkflow(<cloned workflow>)?  Basically changing line 56 to `const selectedWorkflow = {...workflow}`? 

If you do that, then the first point becomes moot... Assuming it's possible.